src/REAXFF/fix_qtpie_reaxff.cpp

```diff
@@ -0,0 +1,1231 @@
+
+  memory->destroy(s_hist);
+  memory->destroy(t_hist);
+
+  FixQtpieReaxFF::deallocate_storage();
+  FixQtpieReaxFF::deallocate_matrix();
+
+  memory->destroy(shld);
+
+  memory->destroy(gauss_exp);
+  if (!reaxflag) {
+    memory->destroy(chi);
+    memory->destroy(eta);
+    memory->destroy(gamma);
+  }
+}
+
+/* ---------------------------------------------------------------------- */
+
+void FixQtpieReaxFF::post_constructor()
+{
+  grow_arrays(atom->nmax);
+  for (int i = 0; i < atom->nmax; i++)
+    for (int j = 0; j < nprev; ++j)
+      s_hist[i][j] = t_hist[i][j] = 0;
+
+  pertype_parameters(pertype_option);
+}
+
+/* ---------------------------------------------------------------------- */
+
+int FixQtpieReaxFF::setmask()
+{
+  int mask = 0;
+  mask |= PRE_FORCE;
+  mask |= PRE_FORCE_RESPA;
+  mask |= MIN_PRE_FORCE;
+  return mask;
+}
+
+/* ---------------------------------------------------------------------- */
+
+void FixQtpieReaxFF::pertype_parameters(char *arg)
+{
+  const int nlocal = atom->nlocal;
+  const int *mask = atom->mask;
+  const int *type = atom->type;
+  const int ntypes = atom->ntypes;
+
+  // read gaussian orbital exponents
+  memory->create(gauss_exp,ntypes+1,"qtpie/reaxff:gauss_exp");
+  if (comm->me == 0) {
+    gauss_exp[0] = 0.0;
+    try {
+      TextFileReader reader(gauss_file,"qtpie/reaxff gaussian exponents");
+      reader.ignore_comments = true;
+      for (int i = 1; i <= ntypes; i++) {
+        const char *line = reader.next_line();
+        if (!line)
+          throw TokenizerException("Fix qtpie/reaxff: Incorrect number of atom types in gauss file","");
+        ValueTokenizer values(line);
+
+        if (values.count() != 2)
+          throw TokenizerException("Fix qtpie/reaxff: Incorrect number of values per line "
+                                   "in gauss file",std::to_string(values.count()));
+
+        int itype = values.next_int();
+        if ((itype < 1) || (itype > ntypes))
+          throw TokenizerException("Fix qtpie/reaxff: Invalid atom type in gauss file",
+                                   std::to_string(itype));
+
+        double exp = values.next_double();
+        if (exp < 0)
+          throw TokenizerException("Fix qtpie/reaxff: Invalid orbital exponent in gauss file",
+                                   std::to_string(exp));
+        gauss_exp[itype] = exp;
+      }
+    } catch (std::exception &e) {
+      error->one(FLERR,e.what());
+    }
+  }
+
+  MPI_Bcast(gauss_exp,ntypes+1,MPI_DOUBLE,0,world);
+
+  // define a cutoff distance (in atomic units) beyond which overlap integrals are neglected
+  // in calc_chi_eff()
+  const double exp_min = find_min_exp(gauss_exp,ntypes+1);
+  const int olap_cut = 10; // overlap integrals are neglected if less than pow(10,-olap_cut)
+  dist_cutoff = sqrt(2*olap_cut/exp_min*log(10.0));
+
+  // read chi, eta and gamma
+
+  if (utils::strmatch(arg,"^reaxff")) {
+    reaxflag = 1;
+    Pair *pair = force->pair_match("^reaxff",0);
```

Comment:
I understand this is called in the post constructor. If so, does this fix's behavior change if I defined a reaxff pair style after this fix is defined in the input script?

Reply:
Correct. One gets `ERROR: No reaxff pair style for fix qtpie/reaxff (src/REAXFF/fix_qtpie_reaxff.cpp:243)` if you put `fix qtpie/reaxff` before defining the reaxff pair style. This behaviour is in alignment with `fix qeq/reaxff` and therefore `fix acks2/reaxff`.